router/router.go

```diff
@@ -200,11 +200,15 @@ func New(cfg *config.Configuration, rateConvertor *currency.RateConverter) (r *R
 	// Metrics engine
 	r.MetricsEngine = metricsConf.NewMetricsEngine(cfg, openrtb_ext.CoreBidderNames(), syncerKeys, moduleStageNames)
 	shutdown, fetcher, ampFetcher, accounts, categoriesFetcher, videoFetcher, storedRespFetcher := storedRequestsConf.NewStoredRequests(cfg, r.MetricsEngine, generalHttpClient, r.Router)
-	// todo(zachbadgett): better shutdown
-	r.Shutdown = shutdown
 
 	analyticsRunner := analyticsBuild.New(&cfg.Analytics)
 
+	// todo(zachbadgett): better shutdown
+	r.Shutdown = func() {
+		shutdown()
+		analyticsRunner.Shutdown()
+	}
```

Comment:
Is it possible to cover this code with tests? I know a lot of code in this file is uncovered already, so no worries if not.

Reply:
let me see what i can do

Reply:
it might work, but requires a big refactoring effort, as the router itself **is not fully dependency inverted** - that means i have to figure out how to pass a proper mock to test this function early.

ideally, i think we should apply dependency inversion pattern to the initialization process of the router to make it easier to test.

cc @bsardo
 let me know if you think it is necessary as part of the PR to start invert analytic runner and `storedRequestsConf.NewStoredRequests` or we should do it as the prebid server 3.0 ( i am all in refactoring the code for better design pattern )